.github/workflows/codeql.yml

```diff
@@ -0,0 +1,38 @@
+name: "CodeQL"
+
+on:
+  workflow_call:
+    inputs: 
+      config-path:
+        required: true 
+        type: string 
+
+jobs:
+  analyze:
+    name: Analyze
+    runs-on: ${{ (matrix.language == 'swift' && 'macos-latest') || 'ubuntu-latest' }}
+    continue-on-error: true
+    timeout-minutes: ${{ (matrix.language == 'swift' && 120) || 360 }}
```

Comment:
It would be good to link to wherever you may have copied / taken inspiration from, for the code above. We don't (currently) use Swift so I'd be minded to remove these conditionals, but it would still be helpful to know what baseline we're building from here.